### PR TITLE
Fixes #10005: Add documentation about inventory upload on root server

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/10005

### CODE REVIEW

File: 8_reference/05_inventory_workflow.txt [hunk 1]
```diff
@@ -0,0 +1,107 @@
+=== Inventory workflow, from nodes to Root server
+
+One of the main information workflow in a Rudder managed system is the node's
+inventory one.
+
+Node inventories are generated on nodes, are sent to the node policy server (be
+it a Relay or the Root server) up to the Root server, and stored in the
+Rudder data base (thechnicaly an LDAP server), waiting for latter use.
```
Comment (amousset) [inline 8_reference/05_inventory_workflow.txt]: technicaly

File: 8_reference/05_inventory_workflow.txt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+spot and solve a problem on the inventory workflow. Following schema sum up
+the whole process.
+
+image:./images/rudder-inventory-workflow.png[Inventory workflow, from node to
+Root server]
+
+
+==== Processing inventories on node
+
+Inventories are generated dayly during an agent run in the 00:00-06:00 time
```
Comment (amousset) [inline 8_reference/05_inventory_workflow.txt]: daily

File: 8_reference/05_inventory_workflow.txt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+$ rudder agent inventory
+
+----
+
+In details, generating inventory does:
+
+* ask the node policy server for its UUID with an HTTP GET on
+  `https://server/uuid`,
+* generate an inventory by scanning the node hardware and software components,
+* optionnal make a digital signature of the generated inventory file,
```
Comment (amousset) [inline 8_reference/05_inventory_workflow.txt]: optionally

File: 8_reference/05_inventory_workflow.txt [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+* look for inventory / signature pairs:
+** inventories without a corresponding signature file are processed only if
+   they are older than 5 minutes,
+* POST the inventory or inventory+signature pair to the local API of
+  "inventory-endpoint" application on `http://localhost:8080/endpoint/upload/`
+* the API makes some quick checks on inventory (well formed, mandatory fields...)
+  and :
+** if checks are OK, *ACCEPTS* (HTTP code `200`) the inventory,
+** if signature is configured to be mandatory and is missing, or if the
+   signature is not valide, refuses with *UNAUTHORIZED* error (HTTP code `401`)
```
Comment (amousset) [inline 8_reference/05_inventory_workflow.txt]: valid

File: 8_reference/05_inventory_workflow.txt (hunk 1 above)
Comment (armeniaca) [inline 8_reference/05_inventory_workflow.txt]: + database

File: 8_reference/05_inventory_workflow.txt (hunk 2 above)
Comment (armeniaca) [inline 8_reference/05_inventory_workflow.txt]: I'dd add "local to the node", y'know, timezones ;-)

File: 8_reference/05_inventory_workflow.txt [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+* the inventory is received by a `webdav` endpoint,
+* the `webdav` service store the file in the folder
+  `/var/rudder/inventories/incoming`
+
+Then, on each run, the agent:
+
+* look for inventory / signature pairs:
+** inventories without a corresponding signature file are processed only if
+   they are older than 5 minutes,
```
Comment (armeniaca) [inline 8_reference/05_inventory_workflow.txt]: this is actually <2 minutes (https://github.com/Normation/rudder-techniques/blob/master/tools/send-clean.sh#L38)

**Comment by fanf** [comment]
PR rebased

File: 8_reference/05_inventory_workflow.txt [hunk 6]
```diff
@@ -0,0 +1,107 @@
+=== Inventory workflow, from nodes to Root server
+
+One of the main information workflow in a Rudder managed system is the node's
+inventory one.
+
+Node inventories are generated on nodes, are sent to the node policy server (be
+it a Relay or the Root server) up to the Root server, and stored in the
+Rudder data base (technically an LDAP server), waiting for latter use.
```
Comment (amousset) [inline 8_reference/05_inventory_workflow.txt]: database, later

**Comment by fanf** [comment]
PR rebased

File: 8_reference/05_inventory_workflow.txt [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+$ curl http://localhost:8080/endpoint/api/info
+
+{
+  "queueMaxSize": 50,
+  "queueFillCount": 50,
+  "queueSaturated": true
+}
+
+----
```
Comment (amousset) [inline 8_reference/05_inventory_workflow.txt]: You need an empty line at the end (to avoid breaking the following title)

**Comment by fanf** [comment]
PR rebased

**Pull request merged**